test/test_optim.py

```diff
@@ -1217,6 +1217,28 @@ def test_no_cyclic_references(self):
         self.assertEqual(
             gc.collect(), 0, msg="Optimizer should be garbage-collected on __del__")
 
+    def test_no_cyclic_references_in_step(self):
+        import gc
+        import weakref
+
+        def run():
+            param = torch.empty(10, requires_grad=True)
+            optim = SGD(params=[param], lr=0.5)
+            scheduler = LambdaLR(optim, lambda epoch: 1.0)
+            param.sum().backward()
+            optim.step()
+            scheduler.step()
+
+            return weakref.ref(scheduler)
+
+        # To ensure that there are no reference cycles in scheduler,
+        # we need to turn off the garbage collector. Since gc will
+        # automatically collect unreachable objects.
+        gc.disable()
+        ref = run()
+        assert ref() is None
+        gc.enable()  # restore
```

Comment:
The `scheduler` object should be freed after `run()` if there are no ref cycles. 

This test ensures that the weakref will be unavailable even if the gc is disabled.  (gc would collect objects with ref cycles)